### ISSUE TITLE
Truemotion usability for block sizes above 16

### ISSUE THREAD

**Issue by Seedmanc** [pinterf/mvtools#16]
Is truemotion even supposed to be used with blocksizes above 16? Whenever I set it to true at 32 (but sometimes 24 and 64) the video gets absolutely destroyed compared to truemotion false or blocksize 16 or below (using mflowfps for example).

Here's a comparison for blocksize 32 at 720p:

![32block Truemotion false](http://puu.sh/Bz51v/97f682fac1.jpg)
truemotion off

![32block Truemotion true](http://puu.sh/Bz54e/db469338ef.jpg)
truemotion on

It doesn't seem to depend on source video type, here's for more 2D-ish looks:

![32block Truemotion false](http://puu.sh/Bz55h/9bc0375b0b.jpg)
truemotion off

![32block Truemotion true](http://puu.sh/Bz55I/513f017a7e.jpg)
truemotion on

At higher source resolutions it starts being noticeable at 24 and 64 blocksizes, at lower - less noticeable at 32. It seems the zigzag nature of the default meander=true is becoming prominent, but nobody ever touches that option, so it's probably not a solution to turn it off.

I assume this is all due to the SAD thresholds/defaults being (still and why?) defined relative to blocksize 8 hence requiring the user to calculate them manually for the new blocksizes. But what should I use for 32? I tried multiplying lsad and lambda by 8 or 16, it didn't help, even if I come up with a combination of parameters that don't look at bad it's still worse by SSIM.

I think that either parameters affected by truemotions should be calculated independent of blocksize (like it's done in [SVP](https://www.svp-team.com/wiki/Manual:SVPflow#Remark:_lambda.2Fplevel_values_meaning.)), or a manual page should provide a clear formula on proper calculation of values for different blocksizes, since it's not obvious now. It would especially make sense for a mod that introduces even larger blocksizes.

**Comment by pinterf** [comment]
SAD thresholds are o.k., they are normalized so that one should use one single value. Internally SAD thresholds are properly converted to whatever blocksize/bitdepth/chroma on/off is used. (base: blocksize 8x8, 8 bit, no chroma).
I'll check the usage of lambda when I have time. The documentation says its value is block size dependent, at least "truemotion" preset sets it so.
Question: are you using 8 bit sources?

**Comment by pinterf** [comment]
And I'd need a short sample clip and the script itself, to talk about the same thing, I cannot see dramatic differences on my samples.

**Comment by Seedmanc** [comment]
Yes, I'm using 8bit sources. I think this problem existed in all versions since 2.5 at least.

As for the samples, the first one is from the standard [Sintel test clip](https://www.youtube.com/watch?v=HOfdboHvshg), starting around 0:30 (lossless available here https://media.xiph.org/). As for the second one, I made a [short excerpt](https://www.dropbox.com/s/e95dh3b46dyetq3/imas.mkv?dl=0). It's in 60fps, I was testing by dropping half the frames and comparing to the original after upsampling.

When upscaling to 2K resolution, it becomes like this at truemotion=true, blocksize 32, overlap 16:
![test](http://puu.sh/BBYNf/f01e8d8c3a.jpg)

**Comment by pinterf** [comment]
Just checking in, I have only a little time nowadays. I could not produce such an ugly artifacts. Could you please provide your exact script (which may be minimized to a level that still exhibits the behaviour), inlcuding the input script resolution, thanks.

**Comment by Seedmanc** [comment]
Okay, so here's the Sintel script where it's quite noticeable:
```
imagesource("f:\480\sintel_trailer_2k_%04d.png",start=1,end=1252)  #images are 480p
converttoyv12
crop(0,58,0,-58)
BicubicResize(1280, 720) #more prominent at higher resolutions

tm=true
bs=32  #32-48
sp=3   # the higher - the more artifacts

super = MSuper( ) 
backward_vec = MAnalyse(super, overlap=bs/2, isb = true,  blksize=bs, truemotion=tm, searchparam=sp ) 
forward_vec  = MAnalyse(super, overlap=bs/2, isb = false,  blksize=bs, truemotion=tm, searchparam=sp )
 
MFlowFps(super, backward_vec, forward_vec, num=2*FramerateNumerator(last), den=FramerateDenominator(last))
selectodd

#frame №716
```

And here's for the imas.mkv I linked above
```
dss2("f:\imas.mkv" ) #720p 60fps
 converttoyv12
 selecteven
 
#BicubicResize(width*2, height*2) 

tm = true
bs=32  #32-48
sp=4  # the higher - the more artifacts, at 10 (720p) or 6 (1440p) starts appearing even for blocksize 64

super = MSuper() 
backward_vec = MAnalyse(super, overlap=bs/2, isb = true,  blksize=bs, truemotion=tm, searchparam=sp ) 
forward_vec  = MAnalyse(super, overlap=bs/2, isb = false,  blksize=bs, truemotion=tm, searchparam=sp )
 
MFlowFps(super, backward_vec, forward_vec, num=2*FramerateNumerator(last), den=FramerateDenominator(last))
selectodd

#frame starting with №11 
```

**Comment by pinterf** [comment]
Thanks, so much better. Previously I've already found internal overflows for larger block sizes, but wasn't sure if it worked, this the exact script helped me to find out where the biggest flaws are.

**Comment by pinterf** [comment]
Released 2.7.32

**Comment by Seedmanc** [comment]
That's much better, thanks.
